client/executor/wasmtime/src/runtime.rs

```diff
@@ -326,6 +327,46 @@ fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config,
 	config.wasm_threads(false);
 	config.wasm_memory64(false);
 
+	let (use_pooling, use_cow) = match semantics.instantiation_strategy {
+		InstantiationStrategy::PoolingCopyOnWrite => (true, true),
+		InstantiationStrategy::Pooling => (true, false),
+		InstantiationStrategy::RecreateInstanceCopyOnWrite => (false, true),
+		InstantiationStrategy::RecreateInstance => (false, false),
+		InstantiationStrategy::LegacyInstanceReuse => (false, false),
+	};
+
+	config.memory_init_cow(use_cow);
+	config.memory_guaranteed_dense_image_size(u64::MAX);
```

Comment:
Could you describe why do you think this is alright, specifically wrt potentially malicious PVFs?